pkg/apis/hive/v1/dnszone_types.go

```diff
@@ -70,6 +74,19 @@ type GCPDNSZoneSpec struct {
 	CredentialsSecretRef corev1.LocalObjectReference `json:"credentialsSecretRef"`
 }
 
+// AzureDNSZoneSpec contains Azure-specific DNSZone specifications
+type AzureDNSZoneSpec struct {
```

Comment:
There should still be an `AzureDNSZoneStatus` with an entry in `DNSZoneStatus`, even though the struct will be empty.